### ISSUE TITLE
Filter Release Date Tied To Specific Type of Release

### ISSUE THREAD

**Issue by halexh** [Fallenbagel/jellyseerr#579]
### Description

Within the Filter section, there is a Release Date range in order to filter Movies that have been released within a certain date range. It would be nice to further specify whether this release date should apply to theatrical release or digital release.

### Desired Behavior

1. Select Movies
2. Select Filters
3. Select Date Range
4. Select some type of checkbox implying this range is for a digital release
5. Filter movies that have had a digital release within the supplied date range

### Code of Conduct

- [X] I agree to follow Jellyseerr's Code of Conduct

**Comment by PierrePetit** [comment]
First of all, thanks for this amazing tool !

To add on what @halexh said :
This would be great to help people find movies that are available straight away. Some of my users are confused as for the availability of the movies they ask for.

A great first step would be to add the filter mentioned. I think also adding a category for "New Releases (At Home)" (and maybe "New Releases (Theaters)") in the "Discover" section would be a great UX improvement, to quickly find something to watch.

Furthermore, adding the different release dates (theatrical, premiere, physical, digital...) on a movie's page would be fantastic. These data are available on TMDB (even though, I have to admit, it's not the most regularly updated field).
There currently is a "status" field in jellyseerr's movies, maybe it would be possible to specify "released (Theaters)" or "released (at Home)" with subcategories as per TMDB's fields :

![image](https://github.com/Fallenbagel/jellyseerr/assets/4937633/8fbbf05f-2eb7-42b7-9e2e-a39ca764e212)

In my view, it would be the most useful update, with the OIDC one, but probably a bit simpler to make 😄 
Thanks for reading.

**Comment by Bretterteig** [comment]
Love @PierrePetit recommendation. I'd add that it would be great to have this be configurable as server default similar to region and language.

**Comment by Fallenbagel** [comment]
> I just saw that some movies had a digital release date in jellyseerr (ex: the three musketeers : D'Artagnan (2023)). The digital release date is an information available on rotten tomatoes. Maybe, adding TMDB's data on top of rottentomatoe's ones could be better ? Rottentomatoes really rarely has this information.

For your information, we don't use rotten tomatoes to get the release dates. Rotten tomatoes are only used to get the ratings. And that's it.

Everything else is tmdb. Including the digital release date of your mentioned movie which also comes from tmdb.

![Screenshot_20240525_150859_Firefox Beta.jpg](https://github.com/Fallenbagel/jellyseerr/assets/98979876/b2575d61-ef88-4152-9a09-33e01d6c29b1)

**Comment by jlato89** [comment]
I would like something like this that instead of thew status being "released" it could instead be more specific like theater release or physical/digital release. Makes it more easy to understand what version is actually released.

**Comment by KristjanKruusRIA** [comment]
+1. would love to get a filter for digital releases.

**Comment by fallenbagel** [comment]
> I would like something like this that instead of thew status being "released" it could instead be more specific like theater release or physical/digital release. Makes it more easy to understand what version is actually released.

I mean technically it is shown
![Screenshot_20260328_153400_Chrome.jpg](https://github.com/user-attachments/assets/55340c45-dae1-4651-a70c-902738eae4b1)

As in you can see the three different release dates and the user can when the current date is past that, then that means its already released

**Comment by demigodmode** [comment]
Id like to take a look at this if nobody’s already working on it.

Thinking of keeping the first pass pretty small: add release-type-aware filtering for movie discovery/search, starting with digital releases from TMDB release type 4. I would probably avoid request gating, admin restrictions and calendar changes for now so the PR doesn’t get too broad

Hope that scope sound useful

**Comment by demigodmode** [comment]
Quick update on this. Ive got a first pass implementation working. Added a new movie discover filter for release type: theatrical (limited)/ theatrical / digital / physical / TV) but before I open the PR I ran into something worth flagging.

Turns out `with_release_type` on TMDB's discover endpoint is basically a no-op unless it's paired with a `release_date.gte`/`release_date.lte` range (+ region). I tested this directly against TMDB's live API with no date range, with_release_type set to every value from 2 through 6 all returned byte identical result sets. Once I added a release_date range + region, the filter actually worked and results change showed meaningful results like 24k vs 10k vs 3.7k total results depending on type

Problem is Seerr's existing **Release Date** filter runs on `primary_release_date.gte`/.lte, a completely different TMDB field. 

Also TMDB's forum says not to combine the two ([discover API - region & with_release_type](https://www.themoviedb.org/talk/65d3043a4d67910163d36437), [with_release_type parameter in search](https://www.themoviedb.org/talk/646cef9b51e6ab013ab97dc3)). So just bolting with_release_type onto the query as is, which is what I did initially, does nothing unless someone also happens to set a matching `release_date` range. whichi doubt anyone would think to do from the UI as it stands. 

Not a TMDB bug exactly but i feel more just an easy to miss quirk in how their API is designed. 

Before I go implement the real fix I wanted to check if that is the right direction or is there a reason Seerr sticks to primary_release_date elsewhere that I should be careful not to break? Dont want to touch shared date filtering logic without a proper check first

**Comment by mawmaw1** [comment]
This would be a great addition to the filtering options. Currently it's very hard to distinguish between films that are available (digitally released) and unavailable (only theatrically released), as the user has to open each movie to figure this out.

**Comment by SoulofAkuma** [comment]
@demigodmode Is this still on your radar? I know this has unfortunately not gotten any response yet from contributors.
If you're not working on this any more, I'm open to take this over.

As a simple solution you could implement for now that would not be too invasive:
- Add a new filter field to the filter slideover to define the minimum release availability
  - The selection should be based on the [options on TMDB](https://developer.themoviedb.org/reference/movie-release-dates) like the Genres multi select already is
  - You can construct the TMDB query filter based on the selection e.g. minimum availability `4=Digital` would indicate to filter TMDB by `with_release_type=4|5|6` 
  - If a value is selected in this field you could default the `Release Date`'s `To` field to today's date if the value is currently empty
- Switch the TMDB query to use `release_date` instead of `primary_release_date` whenever a minimum release availability is selected
  - This is probably the most intuitive solution for most users as long as the filter is just called `Release Date`, as the term is quite generic once you get into the details of releases
 - From my testing no region is required for this to work; the presence of `with_release_type` and `release_date.lte` was sufficient

**Comment by demigodmode** [comment]
@SoulofAkuma yeah still on my radar, appreciate you giving it a nudge. it dropped down my list for a bit but i never dropped it, was mostly waiting on a maintainer to weigh in before i reworked the query.

this is pretty much where i'd already landed earlier, see my [comment ](https://github.com/seerr-team/seerr/issues/579#issuecomment-4887338620). i'd found `with_release_type` does basically nothing on its own and only works once you pair it with a `release_date` range, plus the `primary_release_date` vs `release_date` mismatch. the one bit i wasn't sure on was region. tested it again just now and yeah you're right, no region needed. `with_release_type=4|5|6` + `release_date.lte=<today>` filters fine by itself (got ~314k vs ~1.16M unfiltered), region=US just narrows it more so ill leave that optional.

so plan is to rework my commit along the lines you laid out:

- will drop the single release-type dropdown for the "minimum availability" framing, pick digital and it queries 4|5|6
- default Release Date To to today when an availability is picked and the field's empty
- use release_date instead of `primary_release_date` when availability is set

keeping it to movie discover for the first pass. reworking now, ill get a PR up soon, hopefully. Most of the work is largely complete. 

thanks for the writeup, good to have the direction confirmed.

**Comment by demigodmode** [comment]
edit: 

<img width="613" height="368" alt="Image" src="https://github.com/user-attachments/assets/9b6bcf06-1df8-41ff-80bc-a7694eaa789c" />

**Comment by demigodmode** [comment]
Happy to report, PR is up. But it only partially addresses his issue so I only ref-ed it. :)
